pyanaconda/nm.py

```diff
@@ -961,6 +969,38 @@ def nm_delete_connection(uuid):
     proxy = _get_proxy(object_path=settings_paths[0], interface_name="org.freedesktop.NetworkManager.Settings.Connection")
     proxy.Delete()
 
+
+def nm_update_settings(uuid, new_values):
+    """Update settings of connection given by uuid.
+
+       The type of value is determined from existing settings of device.
+       If setting for key1, key2 does not exist, default_type_str is used or
+       if None, the type is inferred from the value supplied (string and bool only).
+
+       :param uuid: uuid of the connection
+       :type name: str
+       :param new_values: list of settings with new values and its types
+                          [[key1, key2, value, default_type_str]]
+                          key1: first-level key of setting (eg "connection")
+                          key2: second-level key of setting (eg "uuid")
+                          value: new value
+                          default_type_str: dbus type of new value to be used
+                                            if the setting does not already exist;
+                                            if None, the type is inferred from
+                                            value (string and bool only)
+       :type new_values: [[key1, key2, value, default_type_str], ...]
+                         key1: str
+                         key2: str
+                         value:
+                         default_type_str: str
```

Comment:
AFAIK there is no such a place to look at, or I don't know of any (the generated documentation).
But I agree that we could perhaps use NamedTuple here.